frontend/components/branchViewLink.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { SITE } from '../propTypes';
+import appConfig from '../../config/app';
```

Comment:
What about instead using a context variable in the `main` controller to pass explicit config values (both `preview_hostname` and the `templates` config for now) to the rendered template. They could then be exposed as JS in a `<script>` block.